### PR TITLE
chore: update docusaurus to 2.0.0-rc.1

### PR DESCRIPTION
**Pull request by utybo**
I believe there are other issues related 2.0.0-rc.1, but at least it compiles with this PR! 🚀

Edit: I've added a fix for a problem I encountered while trying to use the new version with a website that does *not* use versions. The fix is a complete hack but I'm not aware of any way of making this better. Feel free to edit my PR if you have a better solution for this.

### CODE REVIEW

**Comment by utybo** [comment]
There are still issues when not using versions, I'll probably use the same hack everywhere applicable.

**Comment by utybo** [comment]
*In case people are interested, I have published my fixes as `@utybo/docusaurus-search-local` on [NPM](https://www.npmjs.com/package/@utybo/docusaurus-search-local). I'm currently using it to check that nothing else is broken on several of our (non-public) Docusaurus sites.*

File: docusaurus-search-local/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@easyops-cn/docusaurus-search-local",
-  "version": "0.29.0",
+  "version": "0.30.0",
```
Comment (weareoutman) [inline docusaurus-search-local/package.json:3]: The version will be increased automatically, so we should leave as it is

```suggestion
  "version": "0.29.0",
```

File: docusaurus-search-local/tsconfig.json [hunk 2]
```diff
@@ -5,7 +5,10 @@
     "esModuleInterop": true,
     "importHelpers": true,
     "jsx": "preserve",
-    "moduleResolution": "node",
+    // moduleResolution and module are required because Docusaurus uses 'exports' in its package.json
+    // See here for more information: https://github.com/microsoft/TypeScript/issues/33079#issuecomment-1169377037
+    "moduleResolution": "Node16",
+    "module": "Node16",
```
Comment (weareoutman) [inline docusaurus-search-local/tsconfig.json:11]: 👍 Thanks, I tried upgrade docusaurus to rc.1, but failed, I believe this is the magic here :)

**Review by weareoutman — CHANGES_REQUESTED** [review]

**Comment by weareoutman** [comment]
Thanks! Released in v0.29.1

**Comment by utybo** [comment]
Thank you very much for the fast merge and release! 🚀

**Pull request merged**